connector/spanmetricsconnector/connector.go

```diff
@@ -0,0 +1,544 @@
+package spanmetricsconnector
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+	"unicode"
+
+	"github.com/tilinna/clock"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector/internal/cache"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/traceutil"
+)
+
+const (
+	serviceNameKey     = conventions.AttributeServiceName
+	operationKey       = "operation"   // OpenTelemetry non-standard constant.
+	spanKindKey        = "span.kind"   // OpenTelemetry non-standard constant.
+	statusCodeKey      = "status.code" // OpenTelemetry non-standard constant.
+	metricKeySeparator = string(byte(0))
+
+	defaultDimensionsCacheSize = 1000
+
+	metricNameLatency    = "latency"
+	metricNameCallsTotal = "calls_total"
+)
+
+var defaultLatencyHistogramBucketsMs = []float64{
+	2, 4, 6, 8, 10, 50, 100, 200, 400, 800, 1000, 1400, 2000, 5000, 10_000, 15_000,
+}
+
+type exemplarData struct {
+	traceID pcommon.TraceID
+	spanID  pcommon.SpanID
+	value   float64
+}
+
+type metricKey string
+
+type connectorImp struct {
+	lock   sync.Mutex
+	logger *zap.Logger
+	config Config
+
+	metricsConsumer consumer.Metrics
+	tracesConsumer  consumer.Traces
```

Comment:
We should remove `tracesConsumer`. This connector consumes traces and produces metrics.

Reply:
done